### DISCUSSION TITLE
pipewire - can it handle LiSP?

### DISCUSSION THREAD

**Discussion by mitscherdinger** [FrancescoCeruti/linux-show-player#294] [General]
Hi!
Since pipewire is announced, I'm excited in finally using it with LiSP the way I used it with jackd. Unfortunately, I have failed ever since. Something must be wrong with my setup, although I tried on different machines with different versions.
Is it possible that pipewire can't handle LiSP(, yet, although it's on it's way to 1.0, finally)?
My latest attempt: A desktop machine with a RME HDSP 9652, Ubuntu Studio 23.10, latest cgi-master-version of LiSP (Flatpak), I get an internal data stream error:
"lisp.plugins.gst_backend.gst_utils.GstError: ../libs/gst/base/gstbaseparse.c(3694): gst_base_parse_loop (): /GstPipeline:pipeline0/GstURIDecodeBin:uridecodebin0/GstDecodeBin:decodebin0/GstMpegAudioParse:mpegaudioparse0:
streaming stopped, reason not-linked (-1)"
Could it be, that some permissions are missing? (At installation flatpak told me about needing a permission for "pulseaudio" which is  not on  my system. Didn't told me about anything related to pipewire…)

**Comment by FrancescoCeruti** [comment]
Hi! 
Unfortunately it's an issue with gstreamer and pipewire-jack implementations, at the moment it's unusable.

There's already an open issue (#286) if you want more details

**Comment by mitscherdinger** [comment]
Thanks for the fast response and clarification! Fortunately, at least with ubuntu-studio I can switch to the old pulseaudio/jackd-combo relatively easy. It's more the Gnome-desktop I'm missing…
Just guessing: There is a gstreamer plugin for pipewire… No audio software I'm using offers a native pipewire output. They are all just keeping the "old" pulseaudio, jackd and alsa. I know, pipewire was made to keep compatibility - but it is also a soundserver by its own, isn't it? Is it harder to implement? Or would an implementation just look like alsa/pulseaudio, where you can't configure the exact output to a certain soundcard-channel or a different audio application?

**Reply by FrancescoCeruti** [reply]
There's a GStreamer plugin, provided directly by the PipeWire project. 

I've tried it briefly, without success, I've left it out, for now.

The goal is to support it, and to support routing the cue outputs.
